File: aiplatform-server/src/main/java/com/aieducenter/aiplatform/business/project/application/DispatchStage.java
package com.aieducenter.aiplatform.business.project.application;

/**
 * 派发阶段值集（#50 阶段状态条）：意见 / 咨询全过程阶段帧 {@code dispatch-stage}
 * 的 {@code stage} 值正本（SSE事件清单·通道二）。前端状态条按帧驱动呈现——
 * <b>不署智能体名</b>（分派对用户隐式，阶段进度可见，CONTEXT.md「派发」）。
 *
 * <p>两条链的阶段序：</p>
 * <ul>
 *   <li><b>意见链</b>（已生成项目）：{@code analyzing} →（{@code clarifying} 挂起，
 *       答复即回 {@code analyzing} 续走 / {@code updating-prd} 可选——BA 零动作时
 *       不经此阶，状态条仍完整走完不静默）→ {@code dispatching} 或 {@code queued}
 *       （在途 run 排队如实呈现）→ {@code fixing} → {@code done}（changed 区分
 *       「已修改」与「未动系统」）。生成前意见链止于 BA，不发阶段帧（访谈期以
 *       对话面本身呈现，无隐藏处理段）。</li>
 *   <li><b>咨询链</b>：{@code analyzing} → {@code answered}（零产物短路）。</li>
 * </ul>
 *
 * <p>帧不承担正确性（SSE 事件口径）：失败链唯一的终态阶段帧是
 * {@code dispatch-failed}（BA 收口派修正 run 异常——状态条不悬死在「派发中」），
 * 其余失败无终态阶段帧（error 帧如实表达，状态条停在事发阶段）；重放缓冲可
 * 恢复近期链的阶段（刷新不静默）。</p>
 */
enum DispatchStage {

    /** 分析中：意见 / 咨询已受理，正在分析（BA 轮起跑 / 问答答复续跑同发）。 */
    ANALYZING("analyzing"),

    /** 追问中：BA 挂起提问，链停在等用户答复——答复后续跑回 {@link #ANALYZING}。 */
    CLARIFYING("clarifying"),

    /** 更新 PRD 中：BA 判定需求变更，正在修订 PRD（savePrd 执行）。 */
    UPDATING_PRD("updating-prd"),

    /** 派发中：BA 回合收口，平台正在派修正 run（交接物组装 / 轨道提交）。 */
    DISPATCHING("dispatching"),

    /** 修正中：修正 run 进行中（直播 / 预览随之呈现细节）。 */
    FIXING("fixing"),

    /** 排队中：修正 run 在途，意见已并入下一轮合并修正（如实呈现排队）。 */
    QUEUED("queued"),

    /** 完成：修正收口——changed=true 已修改 / false 未动系统（区分呈现）。 */
    DONE("done"),

    /** 派发失败：BA 收口派修正 run 异常的失败终态——如实告知重提（意见锚已
     * 消费、不自动重试，重提即兜底），状态条不悬死在派发中。 */
    DISPATCH_FAILED("dispatch-failed"),

    /** 已答复：咨询已由平台答复（零产物短路收口）。 */
    ANSWERED("answered");

    /** 线上值（SSE 帧 stage 字段的正本词面，kebab-case）。 */
    private final String wireValue;

    DispatchStage(String wireValue) {
        this.wireValue = wireValue;
    }

    public String wireValue() {
        return wireValue;
    }
}

File: aiplatform-web/src/lib/chat/dispatch-stage.ts
/**
 * 派发阶段帧 → 状态条视图（#50 阶段状态条，纯逻辑单点）：`dispatch-stage` 帧
 * stage 值的名册镜像 + 帧→呈现映射。不署智能体名（分派对用户隐式，阶段进度
 * 可见）；文案遵循「生成」词条 Avoid（不出现「开发」「构建」）。
 *
 * 意见链：analyzing →（clarifying 挂起，答复后续跑 / updating-prd 可选）→
 * dispatching | queued → fixing（接直播——细节在直播侧栏）→ done（changed 区分
 * 「已修改」与「未动系统」）；派发失败落 dispatch-failed 终态（不悬死在派发中，
 * 重提即兜底）。咨询链：analyzing → answered。帧序即阶段序，
 * 项目内最新帧即当前阶段（跨 run：BA 轮锚 → 修正 run 锚，链在项目面上连续）。
 */

/** stage 值名册（镜像服务端 DispatchStage 枚举 wire 值，正本 = SSE事件清单）。 */
export const DISPATCH_STAGES = [
  "analyzing",
  "clarifying",
  "updating-prd",
  "dispatching",
  "fixing",
  "queued",
  "done",
  "dispatch-failed",
  "answered",
] as const;

export type DispatchStage = (typeof DISPATCH_STAGES)[number];

/** 状态条当前态（store 载荷；changed 仅 done 携带）。 */
export type DispatchBarState = {
  stage: DispatchStage;
  changed?: boolean;
};

/** tone 驱动呈现形态：active 转圈进行中 / waiting 等用户或排队 / settled 收口 /
 * failed 失败终态（派发失败如实呈现，非成功收口）。 */
export type DispatchBarTone = "active" | "waiting" | "settled" | "failed";

export type DispatchBarView = {
  /** 用户侧文案（单点正本；无智能体名、无「开发/构建」）。 */
  text: string;
  tone: DispatchBarTone;
};

/** stage 值收窄（payload 信任转型的配套守卫）：名册外 → undefined（忽略不炸）。 */
export function asDispatchStage(value: unknown): DispatchStage | undefined {
  return (DISPATCH_STAGES as readonly string[]).includes(value as string)
    ? (value as DispatchStage)
    : undefined;
}

/** 帧序步进（last-wins：帧序即阶段序，项目内最新帧即当前阶段）。 */
export function nextDispatchBarState(
  prev: DispatchBarState | undefined,
  frame: DispatchBarState,
): DispatchBarState {
  // 同帧重放幂等；新链的 analyzing 直接覆盖上一链的终态
  return prev?.stage === frame.stage && prev.changed === frame.changed ? prev : frame;
}

/** 状态条视图映射（帧 → 文案与形态；无状态 = 不渲染）。 */
export function dispatchBarView(state: DispatchBarState | undefined): DispatchBarView | undefined {
  if (!state) return undefined;
  switch (state.stage) {
    case "analyzing":
      return { text: "正在分析您的意见…", tone: "active" };
    case "clarifying":
      // 挂起停在等用户：问答卡在对话流里，状态条只说「在等」
      return { text: "等待您回答上面的问题，回答后继续处理", tone: "waiting" };
    case "updating-prd":
      return { text: "正在更新 PRD…", tone: "active" };
    case "dispatching":
      return { text: "正在安排修改系统…", tone: "active" };
    case "queued":
      return { text: "已并入下一轮修改，等待当前修改完成", tone: "waiting" };
    case "fixing":
      // 接直播：修正细节在直播侧栏逐段呈现，状态条只持阶段位
      return { text: "正在修改系统…", tone: "active" };
    case "done":
      // 完成态区分（#50）：已修改 / 未动系统（未动原因另见对话流通告）
      return state.changed
        ? { text: "已按您的意见修改了系统", tone: "settled" }
        : { text: "本轮意见未改动系统", tone: "settled" };
    case "dispatch-failed":
      // 派发失败终态（#51）：如实告知重提——意见已消费、不自动重试，重提即兜底
      return { text: "派发失败，请重提您的意见", tone: "failed" };
    case "answered":
      return { text: "已答复您的咨询", tone: "settled" };
  }
}
